src/rrd_parsetime.c

```diff
@@ -832,16 +832,28 @@ static char *day(
 
 static mutex_t parsetime_mutex = MUTEX_INITIALIZER;
 
+static char     *rrd_parsetime_nomt(
+    const char *tspec,
+    rrd_time_value_t * ptv);
+    
 char     *rrd_parsetime(
     const char *tspec,
     rrd_time_value_t * ptv)
 {
-    time_t    now = time(NULL);
-    int       hr = 0;
-
     /* yes this code is non re-entrant ... so lets make sure we do not run
        in twice */
     mutex_lock(&parsetime_mutex);
+    char *result = rrd_parsetime_nomt(tspec, ptv);
+    mutex_unlock(&parsetime_mutex);
```

Comment:
shouldn't you then get rid of the original  unlock line ? (1009 after this patch)